### PR TITLE
Finnished invoice information CRUD

### PR DESCRIPTION
**Pull request by AngelRRuizM**
Missing tests

### CODE REVIEW

File: server/controllers/invoice.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+        let invoice = req.body;
+
+        if(!tutor.tutorDetails.invoiceInformation){
+            tutor.tutorDetails.invoiceInformation = [];
+        }
+
+        let invoiceInformation = tutor.tutorDetails.invoiceInformation
+
+        if(invoiceInformation == null){
```
Comment (Syul968) [inline server/controllers/invoice.js:46]: Same as line 40?

File: server/controllers/invoice.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        let invoices = tutor.tutorDetails.invoiceInformation;
+        
+        let nInvoices = invoices.filter(invoice => invoice._id != invoiceId);
+
+        if(nInvoices.length == invoices.length) {
+            let error = ErrorFactory.buildError(Errors.OBJECT_NOT_FOUND, 'invoice information');
+            return res.status(error.status).send({ error: error });
+        }
+
+        console.log(nInvoices);
```
Comment (Syul968) [inline server/controllers/invoice.js:169]: Do we need this log?

File: server/middleware/validations/invoice-validation.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        'rfc',
+        'invoiceType',
+        'street',
+        'extNum',
+        'intNum',
+        'colony',
+        'country',
+        'state',
+        'municipality',
+        'pc'
```
Comment (Syul968) [inline server/middleware/validations/invoice-validation.js:20]: `postalCode`?

File: server/middleware/validations/invoice-validation.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    ];
+    
+    // Should have study
```
Comment (Syul968) [inline server/middleware/validations/invoice-validation.js]: nit: study?

File: server/middleware/validations/invoice-validation.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        let rfc = invoice.rfc;
+        if(rfc.length < 10){
+            let error = ErrorFactory.buildError(Errors.SHORT_STRING, 'rfc', '10');
+            return res.status(error.status).send({ error: error });
+        }   
+        if(rfc.length > 13){
+            let error = ErrorFactory.buildError(Errors.LONG_STRING, 'rfc', '13');
+            return res.status(error.status).send({ error: error });
+        }  
+        //TODO check reg exp
```
Comment (Syul968) [inline server/middleware/validations/invoice-validation.js:48]: ?

File: server/resources/index.js [hunk 6]
```diff
@@ -3,10 +3,12 @@ const Errors = require('./Errors').Errors;
 
 const locationTypes = ['Espacio publico', 'Casa del tutor', 'Casa del alumno', 'Online'];
 const paymentMethods = ['cash', 'debit card', 'credit card', 'paypal'];
+const invoiceTypes = ['Tipo1', 'Tipo2'];
```
Comment (Syul968) [inline server/resources/index.js:6]: can we use more descriptive types?

**Review by Syul968 — APPROVED** [review]
Looks good

**Comment by RComonfort** [comment]
Please resolve conflicts

**Comment by AngelRRuizM** [comment]
@Syul968 Está bien , solo checala rapido y ya para hacerle merge.

File: seeders/user-seeder.js [hunk 7]
```diff
@@ -11,9 +12,23 @@ const users = [
         phone: 2223454546,
         password: 'password1',
         country: 'Mexico',
-        language: 'Español'
+        language: 'Español',
+        invoiceInformation: [{
+            _id: new mongoose.mongo.ObjectId('5db48a251f3af03783defe7c'),
+            rfc: 'SUSH991111AAA',
+            invoiceType: invoiceTypes[1],
```
Comment (Syul968) [inline seeders/user-seeder.js:19]: 👍

File: server/controllers/invoice.js [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+const ErrorFactory = require('../resources').ErrorFactory;
+const Errors = require('../resources').Errors;
+
+module.exports = {
+    async list(req, res) {
+        let userId = req.params.userId;
+
+        // Validate user exists
+        let user = await User.findById(userId).exec();
+        if(!user ) {
```
Comment (Syul968) [inline server/controllers/invoice.js:14]: nit: space

File: server/controllers/invoice.js [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+        let invoiceInformation = user.invoiceInformation
+
+        if(invoiceInformation == null){
+            invoiceInformation = [];
+        }
+
+        invoice._id = new mongoose.mongo.ObjectId();
+
+        invoiceInformation.push(invoice);
+
```
Comment (Syul968) [inline server/controllers/invoice.js:53]: nit: spacing

File: server/controllers/invoice.js (hunk 2 above)
Comment (Syul968) [reply server/controllers/invoice.js:169]: x2

File: server/middleware/validations/invoice-validation.js [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    // Invoice type validation
+    if(invoice.invoiceType && invoiceTypes.indexOf(invoice.invoiceType) == -1) {
+        let error = ErrorFactory.buildError(Errors.INVALID_FIELD, 'invoice', invoiceTypes);
+        return res.status(error.status).send({ error: error });
+    }
+
+    // Stree max length 255 characters
+    if(invoice.street && invoice.street > 255) {
```
Comment (Syul968) [inline server/middleware/validations/invoice-validation.js:63]: `.length`?

File: test/invoice_information.spec.js [hunk 11]
```diff
@@ -0,0 +1,1024 @@
...
+
+            res.should.have.status(200);
+            res.body.should.be.an('array').that.is.not.empty;
+            
+            done();
+        });
+
+    });
+
+    it('Correct get of no invoices', (done) => {
```
Comment (Syul968) [inline test/invoice_information.spec.js:715]: 👍

**Review by Syul968 — APPROVED** [review]
Awesome! 🐱

**Pull request merged**